gateways/clients/hathor_core_client.py

```diff
@@ -46,8 +47,14 @@ async def get(self, path: str, callback: Callable[[dict], None], params: Optiona
         try:
             async with aiohttp.ClientSession() as session:
                 async with session.get(url, params=params) as response:
+                    self.log.info(
```

Comment:
How does this `await` works? I guess it shouldn't matter to have 2 awaits, right? Or should we have the await first, like this

```
awaited_response = await response

body=awaited_response.text())
callback(awaited_response.json())
```

To be honest I'm not sure how this await works, it's just a question

Reply:
Well, the basic is that any async method returns a coroutine and to get the result of that coroutine you need to `await` it. The response being awaited does not mean that the `text` and `json` methods don't need to be awaited since both of them are async methods.
I think these methods are async because some http response bodies can be chunked, which would make the headers and status of the request available first (status and headers are not async methods) and the body available after waiting for the responder to push the data (which is being waited on asynchronously).